ministryofjustice/opg-paper-identity
# feature/ID-82-Create-PHP-service-to-generate-KBV-questions-and-answers

# Purpose

This PR combines both ID-82, ID-78 & ID-79

To verify functionality:

1 - Create a new case via hitting this endpoint: http://localhost:8001/identity/create

example json payload expected:

`{
	"firstName": "Joe",
    "lastName": "Test",
    "personType": "Donor",
    "dob": "1993-01-09",
    "lpas" : ["M-XOXO-YAGA-XIXL", "M-YOYO-AJAX-3LLP"]
}`

2. Take note of UUID sent back, can also verify this record created here http://localhost:8001/identity/details?uuid=UUID

3. Hit http://localhost:8001/cases/UUID/kbv-questions - this will generate the questions if it does not exist, again can verify this by hitting /details/ endpoint above
 
4. Refresh step 3 to check it is the same questions that is returned back

5. KBV verifications can be tested with any specific case, post generating the questions here: http://localhost:8001/cases/ebdaacbb-f31b-416b-82bc-7ef2ccf709dd/kbv-answers

example json payload expected:

```
{
    "answers": {
        "one": "£5.99",
        "two": "Liberty Trust Bank",
        "three": "July",
        "four": "SafeDrive Insurance"
    }
}
```

You will need to verify what the correct answers to the questions generated for the specific case by referencing https://github.com/ministryofjustice/opg-paper-identity/blob/849a5692ecb7da001cd341f35084334a743ddcfe/service-api/module/Application/src/Mock/KBV/KBVService.php#L52-L52
 

Fixes ###-####

## Approach

_Explain how your code addresses the purpose of the change._

## Learning

_Any tips and tricks, blog posts or tools which helped you._

## Checklist

* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation where relevant
* [ ] I have added tests to prove my work
* [ ] I have run an accessibility tool against the changes and applied fixes
* [ ] The team have tested these changes
